htdocs/core/modules/DolibarrModules.class.php

```diff
@@ -2246,27 +2246,21 @@ public function delete_module_parts()
 		global $conf;
 
 		$err = 0;
-		$entity = $conf->entity;
 
-		if (is_array($this->module_parts) && !empty($this->module_parts)) {
-			dol_syslog(get_class($this)."::delete_module_parts", LOG_DEBUG);
-
-			foreach ($this->module_parts as $key => $value) {
-				// If entity is defined
-				if (is_array($value) && isset($value['entity'])) {
-					$entity = $value['entity'];
-				}
-
-				$sql = "DELETE FROM ".MAIN_DB_PREFIX."const";
-				$sql .= " WHERE ".$this->db->decrypt('name')." LIKE '".$this->db->escape($this->const_name)."_".strtoupper($key)."'";
-				$sql .= " AND entity = ".((int) $entity);
+		$sql = "DELETE FROM ".MAIN_DB_PREFIX."const";
+		$sql .= " WHERE ".$this->db->decrypt('name')." LIKE '".$this->db->escape($this->const_name)."_%'"; // remove all MAIN_MODULE_MYMODULE_*
```

Comment:
This means we don't delete only record stored to save module_part array, but all setup of modules ?
The goal to delete only module_parts is to deleted only constants that are used to enable interactions on code but does not delete all constants that contains data setup. So when we re-enable the module setup is still available.